plugins/modules/deb_publication.py

```diff
@@ -83,6 +91,7 @@ def main():
         argument_spec=dict(
             repository=dict(),
             version=dict(type="int"),
+            mode=dict(default="simple", choices=["simple", "structured", "verbatim"]),
```

Comment:
I think, you can combine simple and structured in the same publication, but maybe we don't realy need this...

cc @quba42

Reply:
It needs to be possible to set both simple and structured for the same APT publication. It is a valid use case.

Reply:
Ack, I'll add parameters for both.

How about verbatim? Should that just be a third parameter?

Reply:
Yes, i think, and it should conflict with the other two.

Reply:
Yes. There are two publishers "APT" and "Verbatim". Of these "APT" has two possible modes: "simple" and "structured". At least one of the two modes must be used. Therefore the full list of different types of possible publications is:
* simple APT
* structured APT
* simple and structured APT
* verbatim
If you add something like "simple and structured" to your list of choices, that should cover every possibility.